### PR TITLE
Fix regression that broke reloads

### PR DESCRIPTION
**Pull request by kgaillot**
This fixes a regression introduced by bdf69722 that broke reloads (no released version is affected).

Additionally there are unrelated minor commits to help static analysis.

### CODE REVIEW

File: tools/crm_mon.c [hunk 1]
```diff
@@ -4439,8 +4444,8 @@ clean_up(crm_exit_t exit_code)
             fprintf(stdout, "Content-Type: text/plain\n"
                             "Status: 500\n\n");
         } else {
-            crm_help('?', CRM_EX_USAGE);
+            return crm_help('?', CRM_EX_USAGE);
         }
     }
-    crm_exit(exit_code);
+    return crm_exit(exit_code);
 }
```
Comment (jnpkrn) [inline tools/crm_mon.c]: `__attribute__((noreturn))` is a typical solution to this;
note that semantically the same is also standardized in C11,
so hopefully the static analyzers are keeping up.

Comment (kgaillot) [reply tools/crm_mon.c]: good point

**Comment by jnpkrn** [comment]
But you still merged the version with superfluous `return`s?

**Comment by kgaillot** [comment]
Yes, they still are useful for readability, and more importantly, they require no extra expenditure of time. We can add proper function attributes whenever we work on the relevant functions, after defining appropriate macros to handle compiler support.

**Comment by jnpkrn** [comment]
It's quite contrary, it misleads regarding the control flow, see also
poorman's optical compensation:
https://github.com/ClusterLabs/pacemaker/commit/68a6e5b92b5a73dd87da36d75534b94eb124a2d1

**Pull request merged**